test/repository.spec.js

```diff
@@ -385,6 +385,17 @@ describe('Repository', function() {
          }));
       });
 
+      it('should successfully write to repo when not providing optional options argument', function(done) {
+         remoteRepo.writeFile('master', fileName, initialText, initialMessage, undefined, assertSuccessful(done, function() {
```

Comment:
I think the reason people are running into this bug is because they're using Promises. The 2 option-less calls I'm imagining people are making:
1. Using callbacks:
``` javascript
remoteRepo.writeFile('master', fileName, initialText, initialMessage, assertSuccessFunction)
```

2. Using Promises:
``` javascript
remoteRepo.writeFile('master', fileName, initialText, initialMessage).then(res => {})
```

I'd like tests to verify the Promise approach, as we have many testing writing files with no options provided, but with a callback.

Reply:
True. Thinking back, I stumbled upon this using async/await. If one were to use the callback approach this would work fine since there is a check in the beginning to see if `options` is a function and in that case use that as the callback.